hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -797,8 +797,10 @@ public OutputStream createFile(final Path path, final FileSystem.Statistics stat
         );
 
       } else {
-        if (getPrefixMode() == PrefixMode.BLOB) {
-          boolean isNormalBlob = !checkIsBlobOrMarker(metadata);
+        boolean isNormalBlob = !checkIsBlobOrMarker(metadata);
+        if (!isNormalBlob && !abfsConfiguration.shouldMkdirFallbackToDfs() && getPrefixMode() == PrefixMode.BLOB) {
+          op = createPathBlob(relativePath, isNormalBlob, overwrite, metadata, null, tracingContext);
+        } else if (!abfsConfiguration.shouldIngressFallbackToDfs() && getPrefixMode() == PrefixMode.BLOB) {
```

Comment:
shouldIngressFallbackToDfs hould be superset of shouldMkdirFallbackToDfs?

lets simplitfy it to 
```
if(!shouldMkdirFallbackToDfs(isNormalBlob) && !abfsConfiguration.shouldIngressFallbackToDfs() &&getPrefixMode() == PrefixMode.BLOB)
 ```

```
shouldMkdirToDFS(isNormalBlob) {
if(isNormalBlob) {return false;}
return abfsConfiguration.shouldMkdirToDFS();
}
```

Reply:
lets simplify similar conditions in the store code.

Reply:
This piece of condition is repeated 3 times, can we create a method?